File: tests/correct/data-structures/ms-queue/variants/main0.c
#include <stdlib.h>
#include <stdio.h>
#include <pthread.h>
#include <stdbool.h>
#include <assert.h>
#include <stdatomic.h>
#include <genmc.h>

#include "../main.c"

int main()
{
	int i = 0;
	unsigned int in_sum = 0, out_sum = 0;

	queue = &myqueue;
	num_threads = readers + writers + rdwr;

	init_queue(queue, num_threads);
	for (int j = 0; j < num_threads; j++)
		param[j] = j;
	for (int j = 0; j < writers; j++, i++)
		pthread_create(&threads[i], NULL, threadW, &param[i]);
	for (int j = 0; j < readers; j++, i++)
		pthread_create(&threads[i], NULL, threadR, &param[i]);
	for (int j = 0; j < rdwr; j++, i++)
		pthread_create(&threads[i], NULL, threadRW, &param[i]);

	/* for (i = 0; i < num_threads; i++) { */
	/* 	param[i] = i; */
	/* 	pthread_create(&threads[i], NULL, main_task, &param[i]); */
	/* } */

	/* for (i = 0; i < num_threads; i++) { */
	/* 	in_sum += input[i]; */
	/* 	out_sum += output[i]; */
	/* } */
	/* for (i = 0; i < num_threads; i++) */
	/* 	printf("input[%d] = %u\n", i, input[i]); */
	/* for (i = 0; i < num_threads; i++) */
	/* 	printf("output[%d] = %u\n", i, output[i]); */
	/* if (succ1 && succ2) */
	/* 	assert(in_sum == out_sum); */
	/* else */
	/* 	assert(0); */

	return 0;
}

File: lli/runtime-include/rust/rust/tests/correct/data-structures/ms-queue-dynamic/variants/main0.rs
#[path = "../common.rs"]
mod common;
pub use common::*;

use std::convert::TryInto;

pub fn main() {
    let _i: i32 = 0;
    let _in_sum: u32 = 0;
    let _out_sum: u32 = 0;

    let num_threads = READERS + WRITERS + RDWR;
    unsafe { init_queue(&MY_QUEUE, num_threads.try_into().unwrap()) };

    let mut param = 0;
    for i in 0..WRITERS {
        let _h = unsafe { std::thread::spawn_f_args(threadW, param + i) };
    }
    //(0..WRITERS).map(|i| unsafe { std::thread::spawn_f_args(threadW, param + i) });

    param += WRITERS;
    for i in 0..READERS {
        let _h = unsafe { std::thread::spawn_f_args(threadR, param + i) };
    }
    //(0..READERS).map(|i| unsafe { std::thread::spawn_f_args(threadR, param + i) });

    param += READERS;
    for i in 0..RDWR {
        let _h = unsafe { std::thread::spawn_f_args(threadRW, param + i) };
    }
    //(0..READERS).map(|i| unsafe { std::thread::spawn_f_args(threadRW, param + i) });
}